src/firefox/index.js

```diff
@@ -239,8 +244,32 @@ export type UseProfileParams = {
   app?: PreferencesAppName,
   configureThisProfile?: ConfigureProfileFn,
   customPrefs?: FirefoxPreferences,
+  createProfileFinder?: typeof defaultCreateProfileFinder,
 };
 
+export function defaultCreateProfileFinder(userDirectoryPath: string = '') {
```

Comment:
This `defaultCreateProfileFinder` function should have a type annotation of the return value:
```js
export function defaultCreateProfileFinder(userDirectoryPath: ?string): IProfileFinder {
  ...
}
```

Once the return value is "type annotated" correctly, flow will notice some new errors that needs to be fixed, in particular:

- `hasProfileName` needs the type annotation of the return value
```js
  hasProfileName: async (profileName: string): Promise<boolean> => {
    ...
  },
```

- we should only catch the exception of `fs.stat` in the `try...catch` at line 256.
- we should re-throw any exception that as been caught by the above `try...catch` which do not verify the `isErrorWithCode('ENOENT', error)` check
- in the test file `"test.firefox.js"` the fake `getPath` defined at lines 417 and 444 should resolve to a string value:
```diff
diff --git a/tests/unit/test-firefox/test.firefox.js b/tests/unit/test-firefox/test.firefox.js
index da6a612..caa54b8 100644
--- a/tests/unit/test-firefox/test.firefox.js
+++ b/tests/unit/test-firefox/test.firefox.js
@@ -414,7 +414,7 @@ describe('firefox', () => {
           sinon.spy((profile) => Promise.resolve(profile));
         const createProfileFinder = () => {
           return {
-            getPath: () => Promise.resolve(),
+            getPath: () => Promise.resolve('/fake/profilePath'),
             hasProfileName: () => Promise.resolve(true),
           };
         };
@@ -441,7 +441,7 @@ describe('firefox', () => {
           sinon.spy((profile) => Promise.resolve(profile));
         const createProfileFinder = () => {
           return {
-            getPath: () => Promise.resolve(),
+            getPath: () => Promise.resolve('/fake/profilePath'),
             hasProfileName: () => Promise.resolve(true),
           };
         };
```